### ISSUE TITLE
Rich text formatting is not rendered when using the Streaming I/O

### ISSUE THREAD

**Issue by Kylir** [guyonroche/exceljs#409]
Hello there,

In my application I'm building an Excel document row by row and stream each row when it's done. One cell has rich formatting - some words have specific colours. I think I'm doing the right thing but in the Excel document the cell has the value `[Object object]`.
I guess this is the `toString()` of the object instead of the correct rendering.

Here is a little example that is failing for me:

```js
var fs = require('fs');
var exceljs = require('exceljs');

var stream = fs.createWriteStream('richtTextStream.xlsx');
var wb = new exceljs.stream.xlsx.WorkbookWriter({'stream': stream, 'useStyles': true});
var ws = wb.addWorksheet('rich');
var row = ws.getRow(2);
var cell = row.getCell(2);

cell.value = {
  richText: [
    {
      font: {color: {argb: '99E391'}},
      text: 'This part '
    },
    {
      font: {color: {argb: 'FF6565'}},
      text: ' and that one.'
    }
  ]
};

row.commit();
ws.commit();
wb.commit();
```

If I use a *non* streaming workbook and write the file using `wb.xlsx.writeFile()` then I can see the correct formatting.

Any idea what is going on? Is there some missing rendering logic in the Streaming I/O?

Thanks!

**Comment by xilix** [comment]
Seems is need to set up "useShareStrings": true in order to make it work

```js
var fs = require('fs');
var exceljs = require('exceljs');

var stream = fs.createWriteStream('richtTextStream.xlsx');
var wb = new exceljs.stream.xlsx.WorkbookWriter({'stream': stream, 'useStyles': true, 'useSharedStrings': true});
var ws = wb.addWorksheet('rich');
var row = ws.getRow(2);
var cell = row.getCell(2);

cell.value = {
  richText: [
    {
      font: {color: {argb: '99E391'}},
      text: 'This part '
    },
    {
      font: {color: {argb: 'FF6565'}},
      text: ' and that one.'
    }
  ]
};

row.commit();
ws.commit();
wb.commit();
```

make sense as the only way that i can find in the openxml format in order to have different rich text formats in the same cell is by using the sharedString.xml file.

Should useSharedStrings be by default true ?

**Comment by Kylir** [comment]
Thanks for the suggestion! I used this `'useSharedStrings': true` option and I can see rich text formatting.

But it is still **not working**: if you have multiple rows with rich text formatting all the cells will have the value of the first cell.

The following script illustrates what I'm saying:

```js
const fs = require('fs');
const exceljs = require('exceljs');

let stream = fs.createWriteStream('richtTextStream.xlsx');
let wb = new exceljs.stream.xlsx.WorkbookWriter({'stream': stream, 'useStyles': true, 'useSharedStrings': true});
let ws = wb.addWorksheet('rich');

let row = ws.getRow(2);
let cell = row.getCell(2);

cell.value = {
  richText: [
    { font: {color: {argb: '99E391'}}, text: 'This part ' },
    { font: {color: {argb: 'FF6565'}}, text: ' and that one.'}
  ]
};
row.commit();

let row2 = ws.getRow(3);
let cell2 = row2.getCell(2);
cell2.value = {
  richText: [
    { font: {color: {argb: '9199E3'}}, text: 'Something different ' },
    { font: {color: {argb: 'FF6565'}}, text: ' with the same colors.' }
  ]
};
row2.commit();

ws.commit();
wb.commit();
```

The excel document generated will have 2 rows with the same content:

| | |
| --- | --- |
|  | This part and that one |
|  | This part and that one |

**Comment by Kylir** [comment]
One more piece of information:

If I unzip the generated Excel document and I look at the sharedStrings.xml file I can see only one string there:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<sst xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main" count="1" uniqueCount="1"><si><r><rPr><color rgb="99E391"/></rPr><t xml:space="preserve">This part </t></r><r><rPr><color rgb="FF6565"/></rPr><t xml:space="preserve"> and that one.</t></r></si></sst>
```

Not sure how to fix that...

**Comment by ddaydd** [comment]
Hi,
Same, duplicate text with

>     worksheet.getCell('F1').value = {
      richText: [
        {font: {bold: true}, text: 'N° DOSSIER :'},
      ]
    };

    worksheet.getCell('H1').value = {
      richText: [
        {font: {bold: true}, text: 'NOM Projet'},
      ]
    };

**Comment by nicsptl** [comment]
This issue is still not fixed in version 1.0.0 . richText repeated across all worksheets.

**Comment by valentinomicudaj1** [comment]
I ran into the same problem with all the richText cells have the same value as the first richText cell, any workaround or any news of a possible fix? Version: 4.2.0
